Source/Embeddings.Processing/EmbeddingsService.cs

```diff
@@ -191,5 +215,46 @@ bool TryGetRegisteredEmbeddingProcessorForTenant(TenantId tenant, EmbeddingId em
             }
             return true;
         }
+
+        async Task<bool> RejectIfInvalidDefinition(
+            EmbeddingDefinition definition,
+            IReverseCallDispatcher<EmbeddingClientToRuntimeMessage, EmbeddingRuntimeToClientMessage, EmbeddingRegistrationRequest, EmbeddingRegistrationResponse, EmbeddingRequest, EmbeddingResponse> dispatcher,
+            CancellationToken cancellationToken)
+        {
+            _logger.ComparingEmbeddingDefinition(definition);
+            var tenantsAndComparisonResult = await _embeddingDefinitionComparer.DiffersFromPersisted(
+                new EmbeddingDefinition(
+                    definition.Embedding,
+                    definition.Events,
+                    definition.InititalState),
+                cancellationToken).ConfigureAwait(false);
+
+            if (!tenantsAndComparisonResult.Values.Any(_ => !_.Succeeded))
+            {
+                return false;
+            }
+            var unsuccessfulComparisons = tenantsAndComparisonResult
+                .Where(_ => !_.Value.Succeeded)
+                .Select(_ =>
+                {
+                    return (_.Key, _.Value);
+                });
```

Comment:
```suggestion
                .Select(_ => (_.Key, _.Value));
```
Is there something magical with tuples so that we can't just do this instead?

Reply:
Yes, it's something magical :)